src/integrations/prefect-redis/prefect_redis/messaging.py

```diff
@@ -0,0 +1,401 @@
+import asyncio
+import json
+import socket
+import uuid
+from contextlib import asynccontextmanager
+from datetime import timedelta
+from functools import partial
+from typing import (
+    Any,
+    AsyncGenerator,
+    Awaitable,
+    Callable,
+    Optional,
+    TypeVar,
+    Union,
+    cast,
+)
+
+import orjson
+from redis.asyncio import Redis
+from redis.exceptions import ResponseError
+from typing_extensions import Self
+
+from prefect.logging import get_logger
+from prefect.server.utilities.messaging import Cache as _Cache
+from prefect.server.utilities.messaging import Consumer as _Consumer
+from prefect.server.utilities.messaging import Message, MessageHandler, StopConsumer
+from prefect.server.utilities.messaging import Publisher as _Publisher
+from prefect_redis.client import get_async_redis_client
+
+logger = get_logger(__name__)
+
+M = TypeVar("M", bound=Message)
+
+MESSAGE_DEDUPLICATION_LOOKBACK = timedelta(minutes=5)
+
+
+class Cache(_Cache):
+    def __init__(self, topic: Optional[str] = None):
+        self.topic = topic or "default"
+        self._client: Redis = get_async_redis_client()
```

Comment:
In conjunction with my comment about the `create_cache` kwarg, I think we need a more unique default topic name. Here's a suggestion, but there are many other valid options.
```suggestion
    def __init__(self, topic: str = "messaging-cache"):
        self.topic = topic
        self._client: Redis = get_async_redis_client()
```